src/thing_physics.c

```diff
@@ -61,7 +61,14 @@ TbBool thing_touching_flight_altitude(const struct Thing *thing)
 TbBool thing_above_flight_altitude(const struct Thing* thing)
 {
     int floor_height = get_floor_height_under_thing_at(thing, &thing->mappos);
-    return (thing->mappos.z.val > floor_height + 19 * NORMAL_FLYING_ALTITUDE / 17);
+    if ((thing->mappos.z.val - floor_height) > (subtile_coord(2,0) + NORMAL_FLYING_ALTITUDE)) // More than 2 blocks over flying altitude is too high for sure
```

Comment:
why subtile_coord? is it used for z axis at all?

Reply:
yes. It's the size of a cube.

Reply:
ok